scipy/signal/tests/test_signaltools.py

```diff
@@ -1088,13 +1088,22 @@ def test_basic(self):
 
     @pytest.mark.parametrize('dtype', [np.ubyte, np.byte, np.ushort, np.short,
                                        np.uint, int, np.ulonglong, np.ulonglong,
-                                       np.float32, np.float64, np.longdouble])
+                                       np.float32, np.float64])
     def test_types(self, dtype):
         # volume input and output types match
         in_typed = np.array(self.IN, dtype=dtype)
         assert_equal(signal.medfilt(in_typed).dtype, dtype)
         assert_equal(signal.medfilt2d(in_typed).dtype, dtype)
 
+    def test_types_deprecated(self):
+        dtype = np.longdouble
+        with suppress_warnings() as sup:
+            sup.filter(DeprecationWarning)
+            in_typed = np.array(self.IN, dtype=dtype)
+            assert_equal(signal.medfilt(in_typed).dtype, dtype)
+            assert_equal(signal.medfilt2d(in_typed).dtype, dtype)
```

Comment:
```suggestion
        in_typed = np.array(self.IN, dtype=dtype)
        msg = "Using medfilt with arrays of dtype"
        with pytest.deprecated_call(match=msg):
            assert_equal(signal.medfilt(in_typed).dtype, type)
        with pytest.deprecated_call(match=msg):
            assert_equal(signal.medfilt2d(in_typed).dtype, dtype)
```

Reply:
Did not know about `pytest.deprecated_call`, thanks!